test/e2e/scenario/recorder.scenario.ts

```diff
@@ -79,25 +81,512 @@ describe('recorder', () => {
 
         const fullSnapshot = findFullSnapshot(events.sessionReplay[0].segment.data)!
 
-        const fooNode = findNodeWithId(fullSnapshot, 'foo')
+        const fooNode = findElementWithIdAttribute(fullSnapshot, 'foo')
         expect(fooNode).toBeTruthy()
         expect(findTextContent(fooNode!)).toBe('foo')
 
-        const barNode = findNodeWithId(fullSnapshot, 'bar')
+        const barNode = findElementWithIdAttribute(fullSnapshot, 'bar')
         expect(barNode).toBeTruthy()
         expect(barNode!.attributes['data-dd-privacy']).toBe('hidden')
         expect(barNode!.childNodes.length).toBe(0)
 
-        const bazNode = findNodeWithId(fullSnapshot, 'baz')
+        const bazNode = findElementWithIdAttribute(fullSnapshot, 'baz')
         expect(bazNode).toBeTruthy()
         expect(bazNode!.attributes.class).toBe('dd-privacy-hidden baz')
         expect(bazNode!.attributes['data-dd-privacy']).toBe('hidden')
         expect(bazNode!.childNodes.length).toBe(0)
       })
   })
 
+  describe('mutations observer', () => {
+    createTest('record mutations')
+      .withSetup(bundleSetup)
+      .withRumRecorder()
+      .withBody(
+        html`
+          <p>mutation observer</p>
+          <ul>
+            <li></li>
+          </ul>
+        `
+      )
+      .run(async ({ events }) => {
+        await browserExecute(() => {
+          const li = document.createElement('li')
+          const ul = document.querySelector('ul') as HTMLUListElement
+          ul.appendChild(li)
+          document.body.removeChild(ul)
+          const p = document.querySelector('p') as HTMLParagraphElement
+          p.appendChild(document.createElement('span'))
+        })
+
+        await flushEvents()
+
+        expect(events.sessionReplay.length).toBe(1)
+
+        const segment = events.sessionReplay[0].segment.data
+        const fullSnapshot = findFullSnapshot(segment)!
+        const mutations = findAllIncrementalSnapshots(segment, IncrementalSource.Mutation) as Array<{
+          data: MutationData
+        }>
+
+        expect(mutations.length).toBe(1)
+
+        expect(mutations[0].data.adds).toEqual([
+          {
+            nextId: null,
+            parentId: findElementWithTagName(fullSnapshot, 'p')!.id,
+            node: jasmine.objectContaining({ tagName: 'span' }),
+          },
+        ])
+        expect(mutations[0].data.removes).toEqual([
+          {
+            parentId: findElementWithTagName(fullSnapshot, 'body')!.id,
+            id: findElementWithTagName(fullSnapshot, 'ul')!.id,
+          },
+        ])
```

Comment:
IMO, there are a lot of similarities across these tests, we could probably make something to reduce the boilerplate and make them more expressive.

what would you think of having a test API like:
```js
flushSessionReplayEvents(events)  // page object-like API to have acces to events, segment, full snapshot
  .singleRecord() // select record to check and do something similar for filterRecordsByIdAttribute usages
    .expectAdded ({  tag: 'span', parentTag: 'p' }) 
    .expectRemoved({ tag: 'ul', parentTag: 'body' })
    .expectNotAdded({ tag: 'li' })
    .expectNotRemoved({ tag: 'li' })
```

Reply:
I'll try something

Reply:
Ok so I did something to remove the boilerplate. Since I didn't write the original tests I wanted to keep some kind of exhaustive snapshot, so I did not use a `expectAdded`/`expectNotAdded` API, and instead add a helper to verify the whole mutations. Because nodes are referenced in the initial full snapshot or the previous mutations, it makes things a bit complex. Let me know what you think about it!